api/metrics-api-jaxrs/src/main/java/org/hawkular/metrics/api/jaxrs/MetricHandler.java

```diff
@@ -704,40 +433,29 @@ public void findAvailabilityData(@Suspended final AsyncResponse asyncResponse,
 
         AvailabilityMetric metric = new AvailabilityMetric(tenantId, new MetricId(id));
         ListenableFuture<AvailabilityMetric> future = metricsService.findAvailabilityData(metric, start, end);
-        Futures.addCallback(future, new FutureCallback<AvailabilityMetric>() {
-            @Override
-            public void onSuccess(AvailabilityMetric metric) {
-                if (metric == null) {
-                    asyncResponse.resume(Response.ok().status(Status.NO_CONTENT).build());
-                } else {
-                    MetricOut output = new MetricOut(metric.getTenantId(), metric.getId().getName(),
-                        MetricUtils.flattenTags(metric.getTags()), metric.getDataRetention());
-                    List<DataPointOut> dataPoints = new ArrayList<>(metric.getData().size());
-                    for (Availability a : metric.getData()) {
-                        dataPoints.add(new DataPointOut(a.getTimestamp(), a.getType().getText(),
-                            MetricUtils.flattenTags(a.getTags())));
-                    }
-                    output.setData(dataPoints);
 
-                    asyncResponse.resume(Response.ok(output).type(APPLICATION_JSON_TYPE).build());
-                }
-            }
-
-            @Override
-            public void onFailure(Throwable t) {
-                asyncResponse.resume(t);
-            }
+        ListenableFuture<List<Availability>> outputfuture = Futures.transform(future,
+                new Function<AvailabilityMetric, List<Availability>>() {
+                    @Override
+                    public List<Availability> apply(AvailabilityMetric input) {
+                        if (input == null) {
```

Comment:
Here null is used. In another function above, it's `NoResultException` instead. I believe these functions are perfect candidate for usage of Java 8 Optional (`Optional.ofNullable`). Besides, this code could be turned into something as simple as:
```java
 outputfuture = Futures.transform(future, input-> Optional.ofNullable(input).map(input->input.getData()));
```